packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -129,7 +129,8 @@ abstract class TransitionRoute<T> extends OverlayRoute<T> {
   //
   // This situation arises when dealing with the Cupertino dismiss gesture.
   @override
-  bool get finishedWhenPopped => _controller!.status == AnimationStatus.dismissed;
+  bool get finishedWhenPopped => _controller!.status == AnimationStatus.dismissed && !_popFinalized;
+  bool _popFinalized = false;
```

Comment:
nit: separate by one empty line

Reply:
What is _popFinalized guarding against?

Reply:
cupertino back swipe gesture will make finishedWhenPopped return true if user swipe all the way to the right but not lifting its finger which causes the page to be disposed twice

Reply:
Just to double-check: Do we have test cases covering that scenario?

Reply:
yes, I figure this out because an existing test failed without this patch